looker/resource_looker_user_attribute_group.go

```diff
@@ -0,0 +1,196 @@
+package looker
+
+import (
+	"context"
+	"errors"
+	"io"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	sdk "github.com/looker-open-source/sdk-codegen/go/sdk/v4"
+	"github.com/resolutionlife/terraform-provider-looker/internal/conv"
+)
+
+func resourceUserAttributeGroup() *schema.Resource {
+	return &schema.Resource{
+		Description:   "Manages Looker User Attribute Groups",
+		CreateContext: resourceUserAttributeGroupCreate,
+		ReadContext:   resourceUserAttributeGroupRead,
+		UpdateContext: resourceUserAttributeGroupUpdate,
+		DeleteContext: resourceUserAttributeGroupDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
```

Comment:
Looks great Mo :) As the ID of this resource isn't a simple ID from the API you'll need a custom import func to set the state. The import basically sets the state in the `schema.ResourceData` and passes it to the read to set the state correctly.